barretenberg/cpp/src/barretenberg/boomerang_value_detection/graph.cpp

```diff
@@ -895,18 +913,85 @@ std::vector<ConnectedComponent> StaticAnalyzer_<FF, CircuitBuilder>::find_connec
     }
     mark_range_list_connected_components();
     mark_finalize_connected_components();
-    if (!return_all_connected_components) {
-        main_connected_components.reserve(connected_components.size());
-        for (auto& cc : connected_components) {
-            if (!cc.is_range_list_cc && !cc.is_finalize_cc) {
-                main_connected_components.emplace_back(std::move(cc));
-            }
+    mark_process_rom_connected_component();
+#ifndef ANALYZER_VERBOSE_DEBUG
+    main_connected_components.reserve(connected_components.size());
```

Comment:
As a rule, changing definitions shouldn't simply make tests fail. Since you are changing the number of returned connected components, tests can start failing. You need to either document this behavior, or change it.